### PR TITLE
Add linter

### LINKED ISSUE #93: Enforce EOL @ EOF
We should enforce EOL @ EOF as part of the build process so we do not get commit noise.

(end of line @ end of file)

### PR DESCRIPTION
**Pull request by AustinShalit**
Closes #93 

Add the doc8 linter to enforce whitespace.  At some point in the future we should fix the line length violations but for now, I am ignoring them.

### CODE REVIEW

**Comment by AustinShalit** [comment]
Grumble grumble windows grumble. @ThadHouse Any ideas?

**Comment by Daltz333** [comment]
```
source/index.rst:63: D002 Trailing whitespace
```

**Comment by AustinShalit** [comment]
Yea. That white space does not exist on my computer.

**Comment by Daltz333** [comment]
Weird. It showed up fine on my Windows machine.

**Comment by AustinShalit** [comment]
I fixed it.  I had to delete the file and add it again.

File: source/docs/hardware/sensors/index.rst [hunk 1]
```diff
@@ -12,9 +12,8 @@ Sensors
    encoders-hardware
    gyros-hardware
    ultrasonics-hardware
+   encoders-hardware
```
Comment (Daltz333) [inline source/docs/hardware/sensors/index.rst]: Remove. Already existing.

**Pull request merged**